.github/workflows/foreman_plugin_js.yml

```diff
@@ -0,0 +1,132 @@
+---
+name: Foreman Plugin JavaScript
+
+on:
+  workflow_call:
+    inputs:
+      plugin:
+        description: Plugin name
+        required: true
+        type: string
+      plugin_repository:
+        description: Plugin repository to use
+        required: false
+        type: string
+      plugin_version:
+        description: Plugin version to use
+        required: false
+        type: string
+      foreman_version:
+        description: Foreman version to use
+        default: 'develop'
+        required: false
+        type: string
+      foreman_repository:
+        description: Foreman repository to use
+        default: 'theforeman/foreman'
+        required: false
+        type: string
+      matrix_include:
+        description: Additional test matrix combinations to include
+        default: '[]'
+        required: false
+        type: string
+      matrix_exclude:
+        description: Test matrix combinations to exclude
+        default: '[]'
+        required: false
+        type: string
+      environment_variables:
+        description: arbitrary environment variables from caller
+        required: false
+        default: ''
+        type: string
+
+env:
+  NODE_ENV: test
+
+jobs:
+  setup_matrix:
+    name: Setup matrix
+    runs-on: ubuntu-latest
+    outputs:
+      matrix: ${{ steps.build_matrix.outputs.matrix }}
+    steps:
+      - name: Build test matrix
+        id: build_matrix
+        uses: theforeman/gha-matrix-builder@v0
+        with:
+          repository: ${{ inputs.foreman_repository }}
+          ref: ${{ inputs.foreman_version }}
+          include: ${{ inputs.matrix_include }}
+          exclude: ${{ inputs.matrix_exclude }}
+
+  test:
+    name: "Foreman ${{ inputs.foreman_version }} Ruby ${{ matrix.ruby }} and Node ${{ matrix.node }}"
```

Comment:
Why do we need ruby for js tests?

Reply:
Foreman's postinstall script calls `script/npm_install_plugins.js`
https://github.com/theforeman/foreman/blob/4abf925fbd72264e8c316d720c03c1dd75688507/package.json#L19
And that, via `script/plugin_webpack_directories.js` needs Ruby.

Reply:
Mind you, we just ensure Ruby is there, we do not install any gems or anything, so that step is rather quick.

(also, you only need foreman to be even there for *some* plugins that share code with Foreman via `find-foreman`: https://github.com/Katello/katello/blob/master/webpack/.eslintrc.js)

Reply:
We could also skip installing a specific Ruby, given that GHA images come with Ruby pre-installed, but it felt wrong to me as that *could* lead to a version of Ruby that is unsupported by our stack.

Reply:
Thanks for the clarification